### ISSUE TITLE
[BUG] Emojis with multiple codepoints break rich table formatting

### ISSUE THREAD

**Issue by vfosterm-ssh** [Textualize/rich#1821]
<img width="386" alt="image" src="https://user-images.githubusercontent.com/82367571/148678593-39cc6815-1e1f-4698-a01d-06d7edee99d5.png">

*Platform*
MacOS Catalina - iterm2

```
python -m rich.diagnose
python -m rich._windows
pip freeze | grep rich
╭───────────────────────── <class 'rich.console.Console'> ─────────────────────────╮
│ A high level console interface.                                                  │
│                                                                                  │
│ ╭──────────────────────────────────────────────────────────────────────────────╮ │
│ │ <console width=87 ColorSystem.TRUECOLOR>                                     │ │
│ ╰──────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                  │
│     color_system = 'truecolor'                                                   │
│         encoding = 'utf-8'                                                       │
│             file = <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'> │
│           height = 48                                                            │
│    is_alt_screen = False                                                         │
│ is_dumb_terminal = False                                                         │
│   is_interactive = True                                                          │
│       is_jupyter = False                                                         │
│      is_terminal = True                                                          │
│   legacy_windows = False                                                         │
│         no_color = False                                                         │
│          options = ConsoleOptions(                                               │
│                        size=ConsoleDimensions(width=87, height=48),              │
│                        legacy_windows=False,                                     │
│                        min_width=1,                                              │
│                        max_width=87,                                             │
│                        is_terminal=True,                                         │
│                        encoding='utf-8',                                         │
│                        max_height=48,                                            │
│                        justify=None,                                             │
│                        overflow=None,                                            │
│                        no_wrap=False,                                            │
│                        highlight=None,                                           │
│                        markup=None,                                              │
│                        height=None                                               │
│                    )                                                             │
│            quiet = False                                                         │
│           record = False                                                         │
│         safe_box = True                                                          │
│             size = ConsoleDimensions(width=87, height=48)                        │
│        soft_wrap = False                                                         │
│           stderr = False                                                         │
│            style = None                                                          │
│         tab_size = 8                                                             │
│            width = 87                                                            │
╰──────────────────────────────────────────────────────────────────────────────────╯
platform="Darwin"
WindowsConsoleFeatures(vt=False, truecolor=False)
rich==10.12.0
```

**Comment by willmcgugan** [comment]
Yes they do. Getting the display size of those emojis is a challenging problem, but we are working on it.

**Comment by amnuts** [comment]
This was closed, but was it actually fixed?  When reported, the version was 10.12.0, but I'm using 12.6.0 and still get the issue:

![Screenshot 2022-11-29 at 13 19 52](https://user-images.githubusercontent.com/684421/204540415-2a5598e7-a9db-42a2-a597-8ed068c24640.png)
![Screenshot 2022-11-29 at 13 20 05](https://user-images.githubusercontent.com/684421/204540423-fa0746fa-89d8-4dca-95be-79e989f34531.png)

If it was solved, what's would I need to do in order to get correct formatting?
